actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -376,13 +376,11 @@ def multiple_sources_tag_builder(type, sources)
         end
 
         def resolve_image_source(source, skip_pipeline)
-          if source.is_a?(Symbol) || source.is_a?(String)
-            path_to_image(source, skip_pipeline: skip_pipeline)
-          else
+          if source.respond_to? :to_model
             polymorphic_url(source)
+          else
+            path_to_image(source, skip_pipeline: skip_pipeline)
```

Comment:
Something like `resolve_path_to_asset` seems reasonable enough... we're intercepting a subset of calls, but I think internally "asset" is just the general superclass of {image,video,audio,etc}.

Talking of paths, the fact we're using `polymorphic_url` seems unfortunate. I saw https://github.com/rails/rails/pull/30084/files#r131984379: it seems that what *should* happen is that we use path when a host isn't required, but keep the host when it is. I thought that was what `polymorphic_path`+`resolve` was already supposed to do. :confused: